link-verifier/verify-links.py

```diff
@@ -0,0 +1,403 @@
+                except Exception as e:
+                    traceback.print_exc()
+                    use_gh_cache = False
+                main_repo_list[repo][PR_CACHED_KEY] = True
+            if main_repo_list[repo][NUM_IS_KEY] > GITHUB_FETCH_THRESHOLD and main_repo_list[repo][ISSUE_CACHED_KEY] == False:
+                try:
+                    fetch_issues(repo, 'issue', 1000)
+                except Exception as e:
+                    traceback.print_exc()
+                    use_gh_cache = False
+                main_repo_list[repo][ISSUE_CACHED_KEY] = True
+
+
+def main():
+    parser = argparse.ArgumentParser(
+        description='A script to test HTTP links, and all links in Markdown files.',
+        epilog='Requires beautifulsoup4, requests, and termcolor from PyPi. ' +
+               'Optional dependencies: pandoc (to support testing Markdown files), gh (To speed up checking GitHub links)'
+    )
+    parser.add_argument("-F", "--files", action="store", dest="files", nargs='+', help="List of files to test links in.")
+    parser.add_argument("-L", "--links", action="store", dest="links", nargs='+', help="List of links to test.")
+    parser.add_argument("-D", "--exclude-dirs", action="store", dest="exclude_dirs", nargs='+', help="List of directories to ignore.")
+    parser.add_argument("-I", "--include-files", action="store", dest="include_files", nargs='+', help="List of file patterns to search for URLs.")
+    parser.add_argument("-A", "--allowlist-file", action="store", dest="allowlist", nargs='+', help="Path to file containing list of allowed URLs.")
+    parser.add_argument("-n", "--num-processes", action="store", type=int, default=4, help="Number of processes to run in parallel")
+    parser.add_argument("-k", "--keep", action="store_true", default=False, help="Keep temporary files instead of deleting")
+    parser.add_argument("-v", "--verbose", action="store_true", default=False, help="Print all links tested")
+    args = parser.parse_args()
+
+    html_file_list = []
+    broken_links = []
+    file_list = []
+    link_list = []
+
+    # If any explicit files are passed, add them to file_list.
+    if args.files is not None:
+        for file in args.files[0].split(','):
+            file_list.append(file)
+
+    # Obtain list of Markdown files from the repository (along with excluding passed directories).
+    for root, dirs, files in os.walk("./"):
+        # Avoid exclude directories, if passed, from search.
+        if args.exclude_dirs is None or not any(exclude_dir in root for exclude_dir in args.exclude_dirs[0].split(',')):
+            for file in files:
+                if file.endswith(".md"):
+                    file_list.append(os.path.join(root, file))
+
+    if args.verbose:
+        print(file_list)
+
+    # If any explicit links are passed, add them to file_list.
+    if args.links is not None:
+        for link in args.links[0].split(','):
+            link_list.append(link)
```

Comment:
`args.links` is iterable
```suggestion
        link_list = args.links
```